api/Dockerfile

```diff
@@ -81,8 +81,12 @@ RUN apt-get update \
       && rm -rf /var/lib/apt/lists/* \
     && mkdir -p /var/log/openverse_api/openverse_api.log
 
+# Create a non-root user
+RUN useradd --create-home opener
+USER opener
+
 # Copy code into the final image
-COPY . /api/
+COPY --chown=opener . /api/
```

Comment:
I will mention (in dealing with this for a local project) that this is a *non-trivial change* and may have adverse and unexpected effects on any volumes we mount. I suspect for the API this won't be as much of an issue, but it's been an issue for some personal work where I've been trying to use the official Airflow image (which uses the `airflow` user rather than `root`).

Reply:
That is unfortunate 😞 Will put it as a draft for now... and get back to it later.

Reply:
@AetherUnbound the use of `COPY` means that the files are being copied into the container and those copies are being `chown`-ed. I don't think it'll affect the permissions on the host but I'm not that well-versed to say for sure without trying it out first.

Reply:
My warnings mostly affect any bound volumes we have. It looks like for the API, we just have this one: https://github.com/WordPress/openverse-api/blob/3479b8c1c377a0be02480a40239c97876eafa407/docker-compose.yml#L75

I _think_ that might be fine, but it's also possible that the `opener` user won't have read access to those files. If it does and the stack works fine then I think my warnings may have been a bit too aggressive, apologies 😄 I know for the catalog we'll have to take a closer look because we do have bind mounts in production (whereas I don't think we do for the API).

Reply:
Tests passed using the `root` user for output-generating commands, woohoo!!

Reply:
We discussed this approach a bit [on the Make WP Slack](https://wordpress.slack.com/archives/C02012JB00N/p1670944461453309), glad it worked out! 😄